### ISSUE TITLE
Command and Powershell cannot start in Sandboxie, and running outside Sandboxie

### ISSUE THREAD

**Issue by greatsteins** [sandboxie-plus/Sandboxie#2752]
### Describe what you noticed and did

1. Find cmd.exe or Powershell, either through shortcuts or system32. Or running a file with ".bat" extension
2. Right click and launch with Sandboxie
3. Error Popup
SBIE2203 Failed to connect to Sandboxie service: *GUIPROXY_00000001; MsgId: 3 - cmd.exe [00000102]
SBIE2205 Service not implemented: ConsoleInit (C0000080)
4. Close the popup
5. cmd and or Powershell runs now and without the Sandboxie indicator around the command window (the default yellow border)
6. under Task manager, the cmd.exe and or Powershell is under "Sandboxie Service", but it's not listed in Sandboxie control panel.
![cmd](https://user-images.githubusercontent.com/122541831/225271231-091a47ad-028b-4bdd-bcf5-af1e60aead4c.png)

### How often did you encounter it so far?

Everytime when trying to run a command prompt or powershell

### Affected program

Everytime when trying to run a command prompt or powershell

### Download link

Windows 10 Pro 22H2 built-in

### Where is the program located?

Not relevant to my request.

### Expected behavior

Run in Sandboxie

### What is your Windows edition and version?

Windows 10 Pro 22H2 64bit updated to latest

### In which Windows account you have this problem?

A local or Microsoft account without special changes.

### Please mention any installed security software

Kaspersky Internet Security 21.3

### What version of Sandboxie are you running?

Sandboxie Classic 5.62.2 (64-bit)

### Is it a new installation of Sandboxie?

I recently did a new clean installation.

### Is it a regression?

After clean Windows 10 installation and updated.

### In which sandbox type you have this problem?

I only reproduced it with Sandboxie Classic.

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

### Did you previously enable some security policy settings outside Sandboxie?

Quit Kaspersky does not help, it will still prompt a popup.
If I checked run with UAC, it would run inside Sandboxie.

### Sandboxie.ini configuration

```shell
#
# Sandboxie configuration file
#

[GlobalSettings]
Template=7zipShellEx
Template=WindowsRasMan
Template=SynapticsTouchPad
Template=WindowsLive
Template=Edge_Fix
Template=Kaspersky
Template=OfficeLicensing
Template=OfficeClickToRun
FileRootPath=C:\Sandbox\%USER%\%SANDBOX%

[DefaultBox]
ConfigLevel=9
AutoRecover=y
BlockNetworkFiles=y
Template=OpenSmartCard
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=AutoRecoverIgnore
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00FFFF,ttl,6
Enabled=y

[UserSettings_23F60340]
SbieCtrl_UserName= ##my user account name hidden##
SbieCtrl_NextUpdateCheck=1679420632
SbieCtrl_AutoApplySettings=n
SbieCtrl_SettingChangeNotify=n
SbieCtrl_HideWindowNotify=n
SbieCtrl_WindowCoords=530,334,1237,632
SbieCtrl_ActiveView=40021
SbieCtrl_ExplorerWarn=n
SbieCtrl_EnableLogonStart=y
SbieCtrl_EnableAutoStart=y
SbieCtrl_AddDesktopIcon=n
SbieCtrl_AddQuickLaunchIcon=y
SbieCtrl_AddContextMenu=y
SbieCtrl_AddSendToMenu=y
SbieCtrl_EditConfNotify=n
SbieCtrl_ReloadConfNotify=n
SbieCtrl_BoxExpandedView=DefaultBox,g

[g]
Enabled=y
ConfigLevel=9
AutoRecover=y
BlockNetworkFiles=y
Template=OpenSmartCard
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=AutoRecoverIgnore
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00FFFF,ttl,6
```

**Comment by ko25july** [comment]
I'm in the same situation.

**Comment by DavidXanatos** [comment]
I can not reproduce this behavioure very strange,
was it like this always for you ?

**Comment by isaak654** [comment]
This issue cannot be reproduced normally, I would suggest (temporarily) uninstalling Kaspersky Internet Security and report back if it worked.

**Comment by isaak654** [comment]
This issue cannot be reproduced normally, I would suggest (temporarily) uninstalling Kaspersky Internet Security and report back if it worked.

Note: `Template=Edge_Fix` is deprecated, and should be removed from Sandboxie.ini.

**Comment by isaak654** [comment]
This issue cannot be reproduced normally (at least with [5.63.1](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.8.1)):

- I would suggest (temporarily) uninstalling Kaspersky Internet Security and report back whether it has an effect
- Note: `Template=Edge_Fix` is deprecated, and should be removed from Sandboxie.ini

**Comment by isaak654** [comment]
This issue cannot be reproduced normally (at least with [5.63.1](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.8.1)):

- I would suggest (temporarily) uninstalling Kaspersky Internet Security and report back whether it has an effect
- `Template=Edge_Fix` is deprecated, and should be removed from Sandboxie.ini

**Comment by greatsteins** [comment]
> I can not reproduce this behavioure very strange, was it like this always for you ?



> This issue cannot be reproduced normally (at least with [5.63.1](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.8.1)):
> 
> * I would suggest (temporarily) uninstalling Kaspersky Internet Security and report back whether it has an effect
> * `Template=Edge_Fix` is deprecated, and should be removed from Sandboxie.ini

- Uninstalled Kaspersky and rebooted, does the same behavior.

- `Template=Edge_Fix` removed, does the same.



I think I found the issue.

The issue is probably due to "third party language input tool without Admin rights."
I had installed this one:
[https://github.com/Chocobo1/windows-chewing-tsf-build/releases]

The language input tool is installed into the system.

- If I switched to this language tool, the command port would fail to launch in Sandboxie, and runs outside Sandboxie.

- If I change to Microsoft built-in language tool, not necessary English, even with Microsoft Traditional Chinese input, it would launch perfectly even without Admin right without issue in Sandboxie.

I tried other programs that would promt a command port, example:
[https://github.com/aria2/aria2/releases]

1. If I launch this program with the 3rd party Language input tool, without Admin right in Sandboxie, it would crash and then initiate the "running outside Sandboxie" issue.

2. Running step 1. with Admin right in Sandboxie, it runs normally without error poping up.

3. With the following ".vbs" extension code file to run the program without Admin right while switched to 3rd party language input, it runs normally inside Sandboxie.
`CreateObject("WScript.Shell").Run "aria2c.exe --conf-path=aria2.conf",0`

So with this behavior, I suspect the difference is to **"not have the command window showing."**

And what difference between showing and not showing the Command window, I though of the language input.

With step 1. and step 2. behavior, I suspect the issue source from Admin rights.

I tried on other ".bat" files which launch Command prompt, and gives the same behavior as step 1., 2. and 3.

Hope this help on reproducing the issue, and fix about it.

I think the title can be edited, if confirmed the issue source, but I don't know what title to give.

**Comment by offhub** [comment]
When running a console program in the sandbox, for some reason (parent-child issue?) the path "`C:\Windows\System32\config\systemprofile\`" is defined to "`conhost.exe`" as a "`USERPROFILE`". Normally, when run outside the sandbox, "`conhost.exe`" runs under the console program and uses its environmental variables.

https://user-images.githubusercontent.com/6871698/230414363-62667497-52c2-4c35-8483-34bd7b0b2fd6.mp4

---

Since the "`USERPROFILE`" path defined for "`conhost.exe`" cannot be accessed without administrator privileges, access is denied and therefore the console program does not work properly.

https://user-images.githubusercontent.com/6871698/230414410-44d1e36d-94fa-4594-8146-73e8c670e6b3.mp4

---

**Workaround**:
```
mklink /D "\Windows\System32\config\systemprofile\ChewingTextService" "%UserProfile%\ChewingTextService"
```

**Comment by greatsteins** [comment]
> When running a console program in the sandbox, for some reason (parent-child issue?) the path "`C:\Windows\System32\config\systemprofile\`" is defined to "`conhost.exe`" as a "`USERPROFILE`". Normally, when run outside the sandbox, "`conhost.exe`" runs under the console program and uses its environmental variables.
> 
> Since the "`USERPROFILE`" path defined for "`conhost.exe`" cannot be accessed without administrator privileges, access is denied and therefore the console program does not work properly.
> 
> **Workaround**: (Unsandboxed) Run CMD.exe as an Administrator:
> 
> ```
> mklink /D "\Windows\System32\config\systemprofile\ChewingTextService" "%UserProfile%\ChewingTextService"
> ```



The [Windows Chewing](https://chewing.im/download.html) Language input method has another version which had been maintained and updated. 
1. The old one: [Windows Chewing TSF](https://github.com/Chocobo1/windows-chewing-tsf-build/releases)（Windows Vista / 7 / 8 / 10）
2. Updated version: [PIME Chewing](https://github.com/EasyIME/PIME/releases)（Windows Vista / 7 / 8 / 10）

Interestingly, while the old Chewing input method would cause Command prompt to fail, the updated PIME Chewing could start in Command prompt and could type "Chewing" language in Command prompt.

However, while the old Chewing input method can be used in "Sandboxie-sandboxed program" for typing Chewing input except Command prompt, the updated PIME Chewing cannot be used to input "Chewing" language in any other "Sandboxie-sandboxed program" (tested with Microsoft Edge, Microsoft notepad, Notepad++, and many other programs).

The new PIME Chewing uses "Python.exe" and had a persist running launcher "PIME Launcher.exe" in the system. Force-stop it and manually run it in Sandboxie could finally make it input Chewing language in the specific Sandboxie-box, but only for the specific box.

In short:
1. Old Chewing
- Cannot input Chewing language in Command prompt.
- Can input Chewing language in other programs, without the need of Admin rights.

2. New PIME Chewing
- Can input Chewing language in Command prompt.
- Cannot input Chewing language in other programs, even with Admin right checked in Sandboxie.
- Manually run the PIME Chewing inside sandboxie can work only for the specific box, by sacrificing the system-wide input.

I guess to make the new PIME Chewing work, would be another issue rather than the original issue. It is profound that it could work in the Sandboxie-Command prompt but fail in any other programs.
